### ISSUE TITLE
[BUG] Datafactory reflect fails on jobs with existing_cluster_id defined

### ISSUE THREAD

**Issue by KrzysztofDobosz** [databrickslabs/dbx#102]
## Expected Behavior

Steps in ADF pipeline are updated.

## Current Behavior

```
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\ProgramData\Anaconda3\envs\SPA\Scripts\dbx.exe\__main__.py", line 7, in <module>
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\click\core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\click\core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\click\core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\click\core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\click\core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\click\core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\dbx\commands\datafactory.py", line 70, in reflect
    reflector.launch()
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\dbx\commands\datafactory.py", line 229, in launch
    service_name = self._create_linked_service(job_spec)
  File "C:\ProgramData\Anaconda3\envs\SPA\lib\site-packages\dbx\commands\datafactory.py", line 172, in _create_linked_service
    existing_cluster_id=cluster_spec.get("existing_cluster_id"),
AttributeError: 'NoneType' object has no attribute 'get'
```

## Steps to Reproduce (for bugs)

1. Use exemplary job. Change conf/deployment.json to use existing_cluster_id.
2. Run `dbx deploy --files-only --write-specs-to-file=.dbx/deployment-result.json`
3. Run `dbx datafactory reflect --specs-file=.dbx/deployment-result.json --subscription-name *** --resource-group *** --factory-name *** --name ***`

## Context

It's probably enough to replace `cluster_spec` with `job_spec` in line 172.

## Your Environment

* dbx version used: 0.2.1
* Databricks Runtime version: 9.1 LTS

**Comment by renardeinside** [comment]
hi @KrzysztofDobosz , thanks a lot for opening the issue. I'll provide a fix and it will be included in 0.2.2 release.

**Comment by renardeinside** [comment]
resolved in #107
